CMakeLists.txt

```diff
@@ -337,8 +337,17 @@ IF(${Kokkos_ENABLE_Cuda})
     ENDIF()
   ENDIF()
 
-  IF (NOT DEFINED KOKKOSKERNELS_ENABLE_TPL_MAGMA)  
-    SET(KOKKOSKERNELS_ENABLE_TPL_MAGMA ON)
+  IF (DEFINED KOKKOSKERNELS_ENABLE_TPL_MAGMA)
+    IF (KOKKOSKERNELS_ENABLE_TPL_MAGMA)  
+      IF (NOT TPL_ENABLE_MAGMA)
+        MESSAGE( WARNING "KOKKOSKERNELS_ENABLE_TPL_MAGMA is ON but TPL_ENABLE_MAGMA is OFF. Please set TPL_ENABLE_MAGMA:BOOL=ON")
```

Comment:
This really should be a hard error:
```
MESSAGE(FATAL ERROR "You set KOKKOSKERNELS_ENABLE_TPL_MAGMA=ON manually, but TPL_ENABLE_MAGMA is OFF, meaning that you are building without the MAGMA library enabled.  You must enable the MAGMA library.  If you do that, you do not need to set KOKKOSKERNELS_ENABLE_TPL_MAGMA.")
```

Reply:
@mhoemmen is right. This is not warning but an configure error.

Reply:
Oh.... I just looked at the cmake file again. Other TPL libraries are set in the same way with a warning. For consistency, this might be okay.

Reply:
@mhoemmen At the end of configure, it shows TPL_LIST. This will make sure users about the libraries used in KokkosKernels.

Reply:
@kyungjoo-kim wrote:
> For consistency, this might be okay.

Wow, I didn't notice that.  I don't like it, but I'll clear my review.